pennylane/transforms/rz_phase_gradient.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A transform for decomposing RZ rotations using a phase gradient catalyst state.
+"""
+import numpy as np
+
+import pennylane as qml
+from pennylane.queuing import QueuingManager
+from pennylane.tape import QuantumScript, QuantumScriptBatch
+from pennylane.transforms import transform
+from pennylane.typing import PostprocessingFn
+from pennylane.wires import Wires
+
+
+def _binary_repr_int(phi, precision):
+    # reasoning for +1e-10 term:
+    # due to the devision by pi, we obtain 14.999.. instead of 15 for, e.g., (1, 1, 1, 1) pi
+    # at the same time, we want to floor off any additional floats when converting to the desired precision,
+    # e.g. representing (1, 1, 1, 1) with only 3 digits we want to obtain (1, 1, 1)
+    # so overall we floor but make sure we add a little term to not accidentally write 14 when the result is 14.999..
+    return int(np.floor(2**precision * phi / (2 * np.pi) + 1e-10))
+
+
+@QueuingManager.stop_recording()
+def _rz_phase_gradient(
+    phi: float, wire: Wires, angle_wires: Wires, phase_grad_wires: Wires, work_wires: Wires
```

Comment:
I would clarify that the global phase is not considered here but it would be added later